blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -366,6 +367,40 @@ Blockly.ContractEditor.prototype.addExampleBlockFromMainBlockSpace = function(ex
 };
 
 /**
+ * Updates examples to allow any type of input.
+ * Used to avoid disconnection in the middle of a type switch-over.
+ * @private
+ */
+Blockly.ContractEditor.prototype.updateExamplesToAnyType_ = function () {
+  this.updateExampleInputsToType_(Blockly.BlockValueType.NONE);
+};
+
+/**
+ * Update examples' input types to match the function definition block's type
+ * @private
+ */
+Blockly.ContractEditor.prototype.updateExampleInputTypes_ = function () {
+  this.updateExampleInputsToType_(this.currentFunctionDefinitionType_());
```

Comment:
Trying to differentiate this function and the next one seems a little confusing. I wonder if it would be cleaner to combine the two:

```
Blockly.ContractEditor.prototype.updateExampleInputTypes_ = function (type_opt) {
  type_opt = (type_opt === "undefined" ? this.currentFuncitonDefinitionType_() : type_opt);
  var blocksToUpdate = this.exampleBlocks.concat(this.functionDefinitionBlock);
  blocksToUpdate.forEach(function (exampleBlock) {
    exampleBlock.updateUsageType(newType);
  }, this);
```
Then updateExamplesToAnyType_ can just call updateExampleInputTypes_(Blockly.BlockValueType.NONE)